component/src/main/java/io/siddhi/extension/io/grpc/sink/GrpcSink.java

```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright (c)  2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.siddhi.extension.io.grpc.sink;
+
+import com.google.protobuf.Empty;
+import io.grpc.Channel;
+import io.grpc.stub.AbstractStub;
+import io.grpc.stub.StreamObserver;
+import io.siddhi.annotation.Example;
+import io.siddhi.annotation.Extension;
+import io.siddhi.annotation.Parameter;
+import io.siddhi.annotation.util.DataType;
+import io.siddhi.core.exception.ConnectionUnavailableException;
+import io.siddhi.core.exception.SiddhiAppCreationException;
+import io.siddhi.core.util.snapshot.state.State;
+import io.siddhi.core.util.transport.DynamicOptions;
+import io.siddhi.core.util.transport.OptionHolder;
+import io.siddhi.extension.io.grpc.util.GrpcConstants;
+import io.siddhi.query.api.exception.SiddhiAppValidationException;
+import org.apache.log4j.Logger;
+import org.wso2.grpc.Event;
+import org.wso2.grpc.EventServiceGrpc;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.concurrent.TimeUnit;
+
+import static io.siddhi.extension.io.grpc.util.GrpcUtils.getRPCmethodList;
+
+/**
+ * {@code GrpcSink} Handle the gRPC publishing tasks.
+ */
+@Extension(name = "grpc", namespace = "sink",
+        description = "" +
+                "This extension publishes event data encoded into GRPC Classes as defined in the user input " +
```

Comment:
@niruhan shouldn't this be 'encoded event data' ?

Reply:
No, we encode the data into grpc classes. I.e create objects of grpc classes with the data from events